Assignment2/Q1.java

```diff
@@ -0,0 +1,148 @@
+//The tree created is a binary tree to match the example given but the method does not make any use of binary tree features 
+
+class Q1
+{
+	public static void main(String[] args)
+	{
+		BinaryTree tree = new BinaryTree(16);
+		tree.add(9);
+		tree.add(18);
+		tree.add(19);
+		tree.add(3);
+		tree.add(14);
+		tree.add(1);
+		tree.add(5);
+
+		System.out.println("OK");
+
+		find(tree, 5);
+	}
+
+	public static void find(BinaryTree tree, int valueToFind)
+	{
+		if (tree.root.value == valueToFind)
+		{
+			System.out.println(valueToFind + " is the root - it has no ancestors");
+		}
+		else if(findHelp(tree.root, valueToFind) == 0)
+		{
+			System.out.println(valueToFind + " is not part of the tree");
+		}
+	}
+
+	public static int findHelp(Node consider, int valueToFind)
+	{
+		if(consider == null)
+		{
+			return 0;
+		}
+		if(consider.value == valueToFind) //found the value
+		{
+			return 1;
+		}	
+		else if(consider.left == null & consider.right == null) //leaf 
+		{
+			return 0;
+		}
+		else if(consider.left == null ) //no left node to examine 
+		{
+			int x = findHelp(consider.right, valueToFind);
+			if(x==1)
```

Comment:
Doesn't this print the ancestors backwards (starting from the leaf, not from the root?) I think the question was asking for printing from the root.